cmake/BuildLibCURL.cmake

```diff
@@ -28,7 +28,12 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCMAKE_C_FLAGS=${LIBCURL_CFLAGS}")
 
     # Switch on the static build.
+    #
+    # The BUILD_STATIC_CURL option has no effect, because
+    # BUILD_CURL_EXE iS OFF. It is set for consistency:
+    # our approach is to set all options explicitly.
```

Comment:
Seems I missed this new rule. Ok, let's set *all* options explicitly.
I've found that these options are missed in `cmake/BuildLibCURL.cmake`:

- `RANDOM_FILE:FILEPATH=/dev/urandom`
- `CMAKE_ADDR2LINE:FILEPATH=/bin/addr2line`
- `CMAKE_COLOR_MAKEFILE:BOOL=ON`
- `CMAKE_C_COMPILER_AR:FILEPATH=/bin/gcc-ar-11`
- `CMAKE_C_COMPILER_RANLIB:FILEPATH=/bin/gcc-ranlib-11`
- `CMAKE_C_FLAGS_DEBUG:STRING=-g `
- `CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG `
- `CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG`
- `CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG `
- `CMAKE_DLLTOOL:FILEPATH=CMAKE_DLLTOOL-NOTFOUND `
- `CMAKE_EXE_LINKER_FLAGS:STRING= `
- `CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING= `
- `CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=`
- `CURL_CA_PATH_SET:BOOL=TRUE `
- `CURL_CA_BUNDLE_SET:BOOL=TRUE`
- `CMAKE_VERBOSE_MAKEFILE`
- `CMAKE_TAPI:FILEPATH=CMAKE_TAPI-NOTFOUND`
- `CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=`
- `CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=`
- `CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING= `
- `CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=`
- `CMAKE_STATIC_LINKER_FLAGS:STRING=`
- `CMAKE_SKIP_RPATH:BOOL=NO`
- ` CMAKE_SKIP_INSTALL_RPATH:BOOL=NO`
- `CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=`
- `CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=`
- `CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=`
- `CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING= `
- `CMAKE_SHARED_LINKER_FLAGS:STRING=`
- `CMAKE_READELF:FILEPATH=/bin/readelf`
- `CMAKE_OBJDUMP:FILEPATH=/bin/objdump`
- `CMAKE_OBJCOPY:FILEPATH=/bin/objcopy `
- `CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=`
- `CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=`
- `CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=`
- `CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=`
- `CMAKE_MODULE_LINKER_FLAGS:STRING=`
- `CMAKE_LDAP_LIB:STRING=ldap`
- `CMAKE_LDAP_INCLUDE_DIR:STRING=`
- `CMAKE_LBER_LIB:STRING=lber`
- `CMAKE_INSTALL_SYSCONFDIR:PATH=etc`
- `CMAKE_INSTALL_SHAREDSTATEDIR:PATH=com`
- `CMAKE_INSTALL_SBINDIR:PATH=sbin`
- `CMAKE_INSTALL_RUNSTATEDIR:PATH=`
- `CMAKE_INSTALL_OLDINCLUDEDIR:PATH=/usr/include`
- `CMAKE_INSTALL_MANDIR:PATH=`
- `CMAKE_INSTALL_LOCALSTATEDIR:PATH=var`
- `CMAKE_INSTALL_LOCALEDIR:PATH=`
- `CMAKE_INSTALL_LIBEXECDIR:PATH=libexec`
- `CMAKE_INSTALL_INFODIR:PATH=`
- `CMAKE_INSTALL_DOCDIR:PATH= `
- `CMAKE_INSTALL_DATAROOTDIR:PATH=share`
- `CMAKE_INSTALL_DATADIR:PATH=`
- `CMAKE_INSTALL_BINDIR:PATH=bin`
- `CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=`

I propose to review options above and set them explicitly in `cmake/BuildLibCURL.cmake` or add to a file `extra/curl-options-exclusions.txt` and after your commit we can easily compare output of `cmake -LAH` for different curl versions to know what CMake options were changed between specified versions:

```
$ cmake -LAH | grep "^[A-Z_]\+" | egrep -v '($(cat extra/curl-options-exclusions.txt | tr "\n" "|"))' | sort > curl_options_v1.txt
$ cmake -LAH | grep "^[A-Z_]\+" | egrep -v '($(cat extra/curl-options-exclusions.txt | tr "\n" "|"))' | sort > curl_options_v2.txt
$ diff -u curl_options_v1.txt curl_options_v2.txt 
```

This will significally simplify following curl versions.

Reply:
`cmake -LA` seems too verbose. How about `cmake -L` (skip advanced options)?

Reply:
In `cmake/BuildLibCURL.cmake` we use advanced options too, it means we should check them as well before every bump. Right?
Also, some Curl options marked as advanced (`mark_as_advanced()`, [^1], for example `HTTP_ONLY`, [^2]), but interested for us.

[^1]: https://cmake.org/cmake/help/latest/command/mark_as_advanced.html
[^2]: https://github.com/curl/curl/blob/a79f20d37612fbc1150a01840967b95a210a574d/CMakeLists.txt#L370-L371

Reply:
Hm. It is interesting. I would like to freeze the libcurl specific options. I don't like to list all the standard cmake options like linker flags (except when we really need it). How to formalize this rule?

Reply:
Added list of options in `cmake/curl-options-exclusion.txt` list and updated commit message and PR message.

Reply:
Thanks! May be `curl-excluded-options.txt` will better? Feel free to ignore.

Reply:
Renamed file, this nameis better.